File: service-interface/src/app/dtos/vehicleDto.ts
export class VehicleRevisionDto {
  id?: number;
  vehicleId?: number;
  date?: string;
  km?: number;
  cost?: number;
  description?: string;
}

export class VehicleItpDto {
  id?: number;
  vehicleId?: number;
  date?: string;
  validityMonths?: number;
  expiryDate?: string;
  cost?: number;
}

export class VehicleInsuranceDto {
  id?: number;
  vehicleId?: number;
  date?: string;
  validityMonths?: number;
  expiryDate?: string;
  insurer?: string;
  policyNumber?: string;
  cost?: number;
}

export class VehicleEventDto {
  id?: number;
  vehicleId?: number;
  date?: string;
  type?: string;
  description?: string;
  cost?: number;
}

export class VehicleDto {
  id?: number;
  licensePlate?: string;
  vin?: string;
  make?: string;
  model?: string;
  year?: number;
  color?: string;
  fuelType?: string;
  engineCapacity?: number;
  power?: number;
  currentKm?: number;
  notes?: string;
  photoName?: string;
  userId?: number;
  userName?: string;
  revisions?: VehicleRevisionDto[];
  itpList?: VehicleItpDto[];
  insuranceList?: VehicleInsuranceDto[];
  events?: VehicleEventDto[];
}

File: ServiceApp/src/main/java/com/example/ServiceApp/dto/VehicleDto.java
package com.example.ServiceApp.dto;

import lombok.*;

import java.util.List;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class VehicleDto {
    private Long id;
    private String licensePlate;
    private String vin;
    private String make;
    private String model;
    private Integer year;
    private String color;
    private String fuelType;
    private Integer engineCapacity;
    private Integer power;
    private Integer currentKm;
    private String notes;
    private String photoName;
    private Long userId;
    private String userName;

    private List<VehicleRevisionDto> revisions;
    private List<VehicleItpDto> itpList;
    private List<VehicleInsuranceDto> insuranceList;
    private List<VehicleEventDto> events;
}
